pom.xml

```diff
@@ -246,6 +259,12 @@ THE SOFTWARE.
       <artifactId>test-annotations</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <!-- make sure our dependency tree and all others are clean of the legacy servlet api.  -->
+      <groupId>javax.servlet</groupId>
+      <!-- the old artifactID for the servlet API -->
+      <artifactId>servlet-api</artifactId>
```

Comment:
IIRC this will inherit the version as there is no `<scope>` element or `<optional>` element... but if there was a `<scope>` or a `<optional>` then this would not match your current depMgmt so the version would not be inferred... (though TBH this is an aspect of depMgmt that is poorly documented)